### ISSUE TITLE
TESS processor issue

### ISSUE THREAD

**Issue by albenson-usgs** [usgs-bcb/tir#20]
I think there is a problem with the TESS processor for the SGCN species. Specifically Pat Lineback (FWS) pointed out that several species that ARE listed by FWS have a status of "Species not listed by USFWS". His two examples are:
[Anaxyrus californicus](https://www1.usgs.gov/csas/swap/species_view.html?sciname=Anaxyrus%20californicus)
[Rana draytonii](https://www1.usgs.gov/csas/swap/species_view.html?sciname=Rana%20draytonii)

Both species are indeed listed by FWS. Interestingly in the TIR we do have the listing status but only for the GAP "version" of those species in the TIR and not the SGCN "version". The SGCN version has TESS as false whereas the GAP version has the FWS listing status. I'm not sure why this is happening as they both appear to have the same TSN.
https://gc2.datadistillery.org/api/v1/sql/bcb?q=SELECT%20*%20FROM%20tir.tir%20WHERE%20scientificname%20=%20%27Anaxyrus%20californicus%27
https://gc2.datadistillery.org/api/v1/sql/bcb?q=SELECT%20*%20FROM%20tir.tir%20WHERE%20scientificname%20=%20%27Rana%20draytonii%27

**Comment by albenson-usgs** [comment]
Here is the sgcn_search version of Anaxyrus californicus: https://gc2.datadistillery.org/api/v1/sql/bcb?q=SELECT%20*%20FROM%20sgcn.sgcn_search%20WHERE%20scientificname%20=%20%27Anaxyrus%20californicus%27
Looks like it might be searching on Bufo californicus and that's why it didn't find a match in TESS?

**Comment by albenson-usgs** [comment]
For Rana draytonii it looks like TESS searched on Rana aurora draytonii: https://gc2.datadistillery.org/api/v1/sql/bcb?q=SELECT%20*%20FROM%20sgcn.sgcn_search%20WHERE%20scientificname%20=%20%27Rana%20draytonii%27
I'm not sure why the processor isn't searching TESS based on the accepted TSN?

**Comment by skybristol** [comment]
We need to review the logic for how the TESS processor operates and then rerun everything using the new infrastructure. It looks like I either did something strange in deciding when to use a scientific name, or the process somehow ran at a time before it picked up the accepted TSN from ITIS. We need to formalize the sequence of processing, and this is really where we could use the message queuing system we're going to try to get all this onto.

**Comment by skybristol** [comment]
Similar to other processes, we probably need to break up the FWS process into two parts where we first establish the linkage from an SGCN unique scientific name record with a FWS TESS identify. This is something the GAP species process in the TIR is teaching us. That will let us exercise a number of search patterns, record any uncertainty in the match, and let a human in the loop take steps to verify the match if necessary. We can then run a process over time to keep the information associated with the linked identify up to date in the cache.

**Comment by skybristol** [comment]
I tracked down an issue in the TESS processor where the TSN was not getting picked up as the method to search. I wiped the field and reran the processor for these two species records and now get the appropriate listing status result. We need to go ahead and wipe the TESS data entirely and rerun everything in case there are other issues like this. I think it probably had to do with timing on when these were run, and we need to work the entire process through afresh using new methods developed against MongoDB.
